agentops/langchain_callback_handler.py

```diff
@@ -112,14 +124,17 @@ def on_chain_start(
         metadata: Optional[Dict[str, Any]] = None,
         **kwargs: Any,
     ) -> Any:
-        self.events[run_id] = Event(
-            event_type="chain",
-            init_timestamp=get_ISO_time(),
-            tags=tags,
-            params={**inputs, **kwargs, **
-                    ({} if metadata is None else metadata)},
+        key = "chain_" + str(run_id)
```

Comment:
rather than prepending keys with a type string, we should just have multiple event dictionaries. this is fragile

Reply:
unless I'm missing something that requires this

Reply:
fixed to use an Events class with strong typing